test(conformance): reduce copying strings

overlookmotel opened the pull request:
The conformance tester strips BOM from start of fixture files if it's present. Previously this always involved copying the whole string. Instead remove the BOM in place, and when there's no BOM to remove, avoid copying the string.

Boshen commented:
### Merge activity

* **Apr 24, 12:01 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
test(conformance): reduce copying strings (#10594)

The conformance tester strips BOM from start of fixture files if it's present. Previously this always involved copying the whole string. Instead remove the BOM in place, and when there's no BOM to remove, avoid copying the string.
